lib/spack/spack/test/conftest.py

```diff
@@ -716,6 +716,31 @@ def mutable_mock_env_path(tmpdir_factory):
     spack.environment.env_path = saved_path
 
 
+@pytest.fixture(scope='session')
```

Comment:
I'll change the name. The fixture is in session scope as it is already used in two tests. The first one is:

https://github.com/spack/spack/blob/74852c49918a00b609b95949cfaea5f206e038b0/lib/spack/spack/test/build_environment.py#L223-L224

while the second one is:

https://github.com/spack/spack/blob/74852c49918a00b609b95949cfaea5f206e038b0/lib/spack/spack/test/llnl/util/filesystem.py#L220-L222

Also, as long as tests using this fixture don't modify the files (and currently I see no reason for that) it seemed more efficient to create the tree structure just once and use it for the whole session.

Reply:
I agree that using session scope does not cause a problem in this case. But session scope often causes problems later that can be difficult to understand as resulting from session scope. Since this is just creating a few files, I don't think there is a significant performance issue with doing that multiple times. If the fixture was used extensively I could see that being a problem.